src/main/java/org/assertj/core/util/introspection/GetterIntrospectionStrategy.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2021 the original author or authors.
+ */
+package org.assertj.core.util.introspection;
+
+import static java.util.Arrays.stream;
+import static java.util.Locale.ENGLISH;
+import static java.util.Objects.requireNonNull;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class GetterIntrospectionStrategy implements IntrospectionStrategy {
+
+  private static final GetterIntrospectionStrategy INSTANCE = new GetterIntrospectionStrategy();
+  private static final PropertySupport propertySupport = PropertySupport.instance();
+
+  private static final String GETTER_REGEX = "^(is|get)\\p{Upper}\\w*";
+
+  public static GetterIntrospectionStrategy instance() {
+    return INSTANCE;
+  }
+
+  @Override
+  public Set<String> getMemberNames(Class<?> clazz) {
+    requireNonNull(clazz, "expecting Class parameter not to be null");
+    Set<String> declaredFields = getDeclaredMethodsIgnoringSyntheticAndStatic(clazz);
+
+    Class<?> superclazz = clazz.getSuperclass();
+    while (superclazz != null && !superclazz.getName().startsWith("java.lang")) {
+      declaredFields.addAll(getDeclaredMethodsIgnoringSyntheticAndStatic(superclazz));
+      superclazz = superclazz.getSuperclass();
+    }
+
+    return declaredFields;
+  }
+
+  @Override
+  public Set<String> getMemberNamesAsFields(Class<?> clazz) {
+    return getMemberNames(clazz)
+                                .stream()
+                                .map(s -> s.startsWith("is") ? s.substring(2) : s.substring(3))
+                                .map(s -> s.substring(0, 1).toLowerCase(ENGLISH) + s.substring(1))
+                                .collect(Collectors.toSet());
+  }
```

Comment:
Following up on my `IntrospectionStrategy` comment, I think this is `getMemberNames` and it should introspect all getter methods but not base this on fields.

After all if users want to compare classes according to their getters, we should be not look at fields.

@nach-o-man  I'm not sure I'm explaining it well, let me know if that's clear for you.

Reply:
If I understand correct you want really to ditinct the code rather that have some complex reusable logic. Is that it?

Reply:
I'm not sure to fully understand your question, what I had in mind was to extract the introspection logic in one place.

Reply:
yes, we want the implementation to depend on the `IntrospectionStrategy` interface only.

we might indeed need `getMemberNamesAsFields` to translate the member location/path (at the moment the location is field based, ex: `person.address.street`), maybe we can rename it to `getMemberLocationNames` or something alike, I'm not completely sure (I'll take the PR branch and see if that makes, it's a bit hard to see from the PR diff)

Reply:
Did you get any insites? At the moment I am trying an approach with changing the  assertion class when switching strategy using 3 strategies - fieldsOnly, Getters, Properties (current). Atmo code is not looking well 😞